core-libs/setup/ssr/logger/loggers/default-express-server-logger.ts

```diff
@@ -56,13 +75,44 @@ export class DefaultExpressServerLogger implements ExpressServerLogger {
       });
     }
 
-    return JSON.stringify(object);
+    return isDevMode()
```

Comment:
[QUESTION]

My gut feeling is that it's risky to use `isDevMode` outside of the context of an Angular app. So we should (1) test it well, and (2) really think twice if we don't abuse the public API of Angular.

So I'm curious how it works in practice. 
I wonder, when the actual setting of `dev mode` vs `prod mode` happens inside angular. I believe setting `prod mode` happens [here](https://github.com/SAP/spartacus/blob/9a85d093f68e74d5c9966dcde0bba44ae0d834ac/projects/storefrontapp/src/main.server.ts#L13-L15) in our `main.server.ts` in our storefrontapp. In our storefrontapp we guard this setting with a condition `environment.production` which is based on the `environment.ts` files replacable in build time (thanks to `angular.json` configuration). That said, I vaguely remember that in Angular v15 the core team no longer creates `environment` files for new apps, but uses [some different mechanism for setting production variables](https://dev.to/this-is-angular/angular-15-what-happened-to-environmentts-koh) - I didn't check it yet, so to be invesitaged. So I'm not sure how the setting of 'dev mode' vs 'prod mode' looks in modern Angular 15+ apps -> we also should check it.

I'd like us to test the following cases:
1. remove almost everything from `server.ts` and use `new DefaultEspressServerLogger()` in isolation - try out logging something with some context. Why: to be sure that `main.server.ts` doesn't influence it. Let's test it both with: prod build SSR and dev build SSR. What's the result - inline or multi-line?
2. bring back everything (including an import of AppServerModule to server.ts). Let's create manually `new DefaultEspressServerLogger()` in the top of the `server.ts` and log something very early with some context. Check prod build SSR and dev build SSR. is it multiline or single line? Then let's make request to this SSR to trigger an actual Angular rendering. What's the result inline or multi-line? Especially is the result in a different format than the early log we've put manually in the top of `server.ts`? Why to test it: I'm interested if there can be any "race condition" - e.g. if before the first render of any Angular app the `isDevMode()` would behave differently?

Note: I'm interested if any warnings thrown to console by Angular feamework about too early accessing `isDevMode` or that `dev mode` and then possibly changing this value in a `prod mode` later on, when the first rendering starts. Seeing [Angular source code of `isDevMode` from version 14.2.x](https://github.com/angular/angular/blob/61e69c87e6b577f1c5a71fb5c39c3bd6125c4b91/packages/core/src/util/is_dev_mode.ts#L30-L48), I'd expect that the value of `dev mode` cannot be changed after the first read of this value. However the [Angular source code of `isDevMode` since version 15.0.0](https://github.com/angular/angular/blob/b79ad226b2c6fc2dfaafe992facd7172fa0944c3/packages/core/src/util/is_dev_mode.ts#L20-L22) doesn't seem to have this check.
3. Let's figure out how modern ng15 applications work e.g. by generating aside a fresh ng15.0.0 angular app (since they don't have generated replacable `environment.ts` files). Then let's see how such modern applications setup a `prod mode` - do they do it explicitly as we do in our stoerfront app? Or since ng15 it's configured implicitly and internally by Anuglar CLI when building a project with "prod" or "dev" mode of the build? When we have an answer, it might be worth to repeat the experiment (2), but with storefrontap adapted to have more modern configuration (like in fresh ng15.0.0 app).

Reply:
As we discussed and tested, `isDevMode` can be safely used outside the Angular context.

Reply:
Yea. The reason it can be safely used, is that Angular implicitly sets `ngDevMode` to false, when building app in prod mode (with optimizations enabled) via `terser`. In details:
- in `@angular/compiler-cli` the setting of `ngDevMode` to `false` is exported as a special config object `GLOBAL_DEFS_FOR_TERSER` (and as `GLOBAL_DEFS_FOR_TERSER_WITH_AOT`:
    https://github.com/angular/angular-cli/blob/8089c9388056b3caaf56f981848aca94f022da73/packages/angular_devkit/build_angular/src/tools/webpack/configs/common.ts#L296-L299
  - Then `@angular-devkit/build-angular` picks this config. And when it's building an app with `optimizations: true` (in prod mode), it "hardcodes" those global constants values (`GLOBAL_DEFS_FOR_TERSER` with `ngDevMode` set to `false`), by passing those constants to special config key `"define"` of the options of `Terser` otimizer:
    https://github.com/angular/angular-cli/blob/8089c9388056b3caaf56f981848aca94f022da73/packages/angular_devkit/build_angular/src/tools/webpack/configs/common.ts#L296-L299